crates/ide/src/syntax_highlighting.rs

```diff
@@ -746,6 +746,9 @@ fn highlight_def(db: &RootDatabase, def: Definition) -> Highlight {
                 if func.is_unsafe(db) {
                     h |= HighlightModifier::Unsafe;
                 }
+                if func.is_associated(db) {
```

Comment:
Let's use something like this instead:

```suggestion
                if func.is_assoc_item(db) && func.self_param(db).is_none() {
```

Reply:
Oh dear, that was one confusing comment, sorry. 

What I meant to say is that we already have **as**_assoc_item function we can use here, instead of implementing a new `is_assoc_item`:

https://github.com/rust-analyzer/rust-analyzer/blob/2f247140817c9cbd9009085c9f9ccedb4f6a718f/crates/hir/src/code_model.rs#L1002